### DISCUSSION TITLE
Mqtt5 Client connection failed with error: libaws-c-mqtt: AWS_ERROR_MQTT5_CONNACK_CONNECTION_REFUSED, Remote endpoint rejected the CONNECT attempt by returning an unsuccessful CONNACK

### DISCUSSION THREAD

**Discussion by gotcarriedaway** [aws/aws-iot-device-sdk-cpp-v2#729] [Q&A] [answered]
Getting error  as below:

Mqtt5 Client connection failed with error: libaws-c-mqtt: AWS_ERROR_MQTT5_CONNACK_CONNECTION_REFUSED, Remote endpoint rejected the CONNECT attempt by returning an unsuccessful CONNACK 

and it appears occasionally , is it a network issue or what?

**Comment by bretambrose** [comment, marked as answer]
What is the reason code in the connack packet?

**Reply by jmklix** [reply]
@gotcarriedaway can you provide more details around the connection failure that you're seeing? It will help us further diagnose this issue.
